File: demo_mask.m
%%
% Reads in and plots the region definitions in the demo mask netcdf file.

f ='demo_mask.nc';

% read in some data
t = h5read(f, '/Interpretation/mask_times');
d = h5read(f, '/Interpretation/mask_depths');
mind = h5read(f, '/Interpretation/min_depth');
maxd = h5read(f, '/Interpretation/max_depth');
st = h5read(f, '/Interpretation/start_time');
et = h5read(f, '/Interpretation/end_time');
id = h5read(f, '/Interpretation/id');
name = h5read(f, '/Interpretation/name');

c = h5read(f, '/Interpretation/sound_speed');

% and some attributes
time_units = h5readatt(f, '/Interpretation/mask_times', 'units');
depth_units = h5readatt(f, '/Interpretation/mask_depths', 'units');
c_units = h5readatt(f, '/Interpretation/sound_speed', 'units');

clf
% plot each bounding box and the ping masks inside the boxes
for i = 1:length(t)
    plot([st(i) et(i) et(i) st(i) st(i)],[mind(i) mind(i) maxd(i) maxd(i) mind(i)])
    hold on
    
    text(double(st(i)), double(mind(i)), [num2str(id(i)) ', (' name{i} ')'])

    
    dd = reshape(d{i}, 2, length(d{i})/2)';
    tt = t{i};
    
    for j = 1:length(tt)
        plot([tt(j) tt(j)], [dd(j,1) dd(j,2)], 'LineWidth', 2)
    end
end

title(['Regions (c=' num2str(c) ' [' c_units '])'])

set(gca, 'Ydir', 'reverse')
xlabel(['Time (' time_units ')'])
ylabel(['Depth (' depth_units ')'])

File: demo_mask.py
# -*- coding: utf-8 -*-
"""
Created on Tue Aug 20 11:44:06 2019

@author: gavinj
"""

from netCDF4 import Dataset
import numpy as np
import matplotlib.pyplot as plt

#Name of the netcdf file
filename = 'demo_mask.nc'

#open the file
dataset = Dataset(filename)

#Open the group where the data is located
interp = dataset.groups['Interpretation']

# Get some variables and attributes
t = interp.variables['mask_times'][:]
d = interp.variables['mask_depths'][:]

c = interp.variables['sound_speed'][:]
c_units = interp.variables['sound_speed'].units

t_units = interp.variables['mask_times'].units
d_units = interp.variables['mask_depths'].units

bb_upper = interp.variables['min_depth'][:]
bb_lower = interp.variables['max_depth'][:]
bb_left = interp.variables['start_time'][:]
bb_right = interp.variables['end_time'][:]

region_id = interp.variables['id'][:]

#close dataset
dataset.close()

#Plot the power of beam
plt.figure(1)
plt.clf()

# plot masks
for i, r in enumerate(d):
    plt.plot(np.array([t[i], t[i]]), np.array([r[0::2], r[1::2]]), linewidth=4, color='k')

# plot bounding boxes
for i, junk in enumerate(bb_upper):
    plt.plot(np.array([bb_left[i], bb_right[i], bb_right[i], bb_left[i], bb_left[i]]), 
             np.array([bb_lower[i], bb_lower[i], bb_upper[i], bb_upper[i], bb_lower[i]]),
             color=(0.5, 0.5, 0.5))
    plt.text(bb_left[i], bb_upper[i], 'ID: ' + str(region_id[i]), 
             bbox=dict(facecolor=(0.5, 0.5, 0.5), alpha=0.5))
plt.title('Using c= ' + str(c) + ' ' + c_units)
ax = plt.gca()
ax.invert_yaxis()

plt.xlabel('Time\n(' + t_units + ')')
plt.ylabel('Depth (' + d_units + ')')

plt.savefig('demo_mask.png', bbox_inches='tight')
